viewing_party/party.py

```diff
@@ -1,23 +1,176 @@
 # ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return movie
+    else: 
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    wacthlist = user_data["watchlist"]
+
+    for movie in wacthlist:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    """
+    Calculates the average rating of watched movies.
+
+    Returns:
+    float: The average rating of watched movies, or 0.0 if none are watched.
+    """
+
+    avg_rating = 0.0
+    count = 0
+    sum = 0
+
+    if not user_data["watched"]:
+        return avg_rating
+    
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+        count += 1
 
+    avg_rating = sum / count
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    """
+    Determines the most watched genre among watched movies.
+
+    Returns:
+    str: The most watched genre, or None if no movies have been watched.
+    """
+
+    movie_genre = {}
+    count = 0
+    genre = ""

```

Comment:
Prefer `genre` to be more descriptively named to reflect that this isn't just any genre, but it's the most watched genre. Maybe something like `fave_genre` or something to that effect.